generators/app/index.js

```diff
@@ -46,11 +46,13 @@ module.exports = class extends Generator {
 			this.bluemix = {};
 			this.opts.bluemix = this.bluemix;
 		}
+
+		this.cloudDeploymentType = this.bluemix.server && this.bluemix.server.cloudDeploymentType;
```

Comment:
I believe this will return a bool result, did you mean to use `||` instead of `&&`?